extensions/cl_img_swap_ops.asciidoc

```diff
@@ -0,0 +1,122 @@
+:data-uri:
+:icons: font
+include::../config/attribs.txt[]
+:source-highlighter: coderay
+
+= cl_img_swap_ops
+
+== Name Strings
+
+`cl_img_swap_ops`
+
+== Contact
+
+Imagination Technologies Developer Forum: +
+https://forums.imgtec.com/
+
+Tomasz Platek, Imagination Technologies (Tomasz.Platek 'at' imgtec.com)
+
+== Contributors
+
+CY Cheng, Imagination Technologies. +
+Tomasz Platek, Imagination Technologies.
+
+== Notice
+
+Copyright (c) 2024 Imagination Technologies Ltd. All Rights Reserved.
+
+== Status
+
+Final Draft
+
+== Version
+
+Built On: {docdate} +
+Version: Major.Minor.Patch
+
+== Dependencies
+
+This extension is written against the OpenCL C Specification Version V3.0.16.
+
+== Overview
+
+This extension adds built-in functions that exercise hardware capabilities of Imagination GPU IP and expose cross work-items swap functions.
+
+== New OpenCL C Feature Names
+
+[source,c]
+----
+__opencl_img_swap
+----
+
+== New OpenCL C Functions
+
+Perform the swap operation:
+
+[source,c]
+----
+gentype img_swap_x(gentype value);
+gentype img_swap_y(gentype value);
+----
+
+== Modifications to the OpenCL C Specification
+
+(Add to Table 16 - Built-in Scalar and Vector Argument Common Functions in Section 6.15.4 - Common Functions) ::
++
+--
+[cols="1,2",options="header"]
+|====
+| Function | Description
+| gentype *img_swap_x*(gentype value)
+    a| `img_swap_x` swaps `values` between work-items in the following way: all work-items are divided into blocks of four consecutive elements. For each block: 
```

Comment:
Consider documenting the behavior if the number of work-items is not evenly divisible by four, or if some work-items in the block of four are inactive (due to flow control).

Also consider documenting what "consecutive" means, especially for multi-dimensional work-groups.

Reply:
I defined the behavior as undefined for these cases as I don't see any reasonable way of handling them.